shielder/contract/lib.rs

```diff
@@ -123,6 +120,18 @@ pub mod contract {
             Ok(())
         }
 
+        /// Returns merkle root of notes storage
+        #[ink(message)]
+        pub fn notes_merkle_root(&self) -> Scalar {
+            self.notes.root()
+        }
+
+        /// Returns merkle path
+        #[ink(message)]
+        pub fn notes_merkle_path(&self, note_id: u32) -> Result<[Scalar; DEPTH], ShielderError> {
+            self.notes.gen_proof(note_id)
+        }
```

Comment:
yeah, this is exactly the way of fetching info that might bit problematic (revealing to the RPC node/network your exact position in the tree)

I think we can leave it now as it is and try to come up with mechanism for fetching whole tree (or a big part)